### DISCUSSION TITLE
Locale vs Language

### DISCUSSION THREAD

**Discussion by xavipolo** [sergiodxa/remix-i18next#101] [Q&A] [answered]
Hello,

what is the best way to get the browser locale?

My application is translated to a small number of languages, but I want to be able to offer the possibility to show dates and numbers with the format that the user wants, for example my app is not translated to japanese but it must support to show dates and numbers in that language.

With `i18n.options.supportedLngs` I have the list of configured languages, with `i18nextServer.getLocale(request)` the language detected by i18next from all the configured ones, but is there an "efficient" way to get the list of locales obtained by the language detector?

My idea is to define it giving preference to the user's settings or to what is configured in the browser.

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
The `getLocale` method will give you the user locale that the user wants from your supported list of languages.

So if the user prefers Spanish over English, and you support both the `getLocale` function will give you `es`.

If you want the whole list of detected locales you'll have to copy this https://github.com/sergiodxa/remix-i18next/blob/main/src/lib/get-client-locales.ts which is what the language detector class uses internally to get the user locales from the Accept-Language header

**Reply by xavipolo** [reply]
ok, 
thanks!
